orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/monitoreddeploy/MonitoredDeployBaseTask.java

```diff
@@ -132,10 +190,60 @@ private TaskResult handleError(
         monitorDefinition,
         e);
 
-    return TaskResult.builder(ExecutionStatus.SUCCEEDED)
-        // TODO(mvulfson)
-        // .context()
-        .build();
+    String userMessage =
+        String.format(
+            "Failed to get a valid response from deployment monitor %s, proceeding anyway because this deployment monitor is configured to not failOnError",
+            monitorDefinition.getName());
+
+    return buildTaskResult(TaskResult.builder(ExecutionStatus.SUCCEEDED), userMessage);
+  }
+
+  private void ensureMonitorDefinition(Stage stage) {
+    if (this.stage == null) {
+      MonitoredDeployStageData context = stage.mapTo(MonitoredDeployStageData.class);
+      this.stage = stage;
+      this.monitorDefinition =
+          deploymentMonitorServiceProvider.getDefinitionById(
+              context.getDeploymentMonitor().getId());
+    }
+  }
+
+  TaskResult buildTaskResult(
+      TaskResult.TaskResultBuilder taskResultBuilder, EvaluateHealthResponse response) {
+    List<StatusReason> statusReasons =
+        Optional.ofNullable(response.getStatusReasons()).orElse(Collections.emptyList());
+
+    String summary = "Deployment monitor requested to: " + response.getNextStep().getDirective();
+    StatusExplanation explanation = new StatusExplanation(summary, statusReasons);
+
+    return taskResultBuilder.context("deploymentMonitorReasons", explanation).build();
+  }
+
+  TaskResult buildTaskResult(TaskResult.TaskResultBuilder taskResultBuilder, String summary) {
+    StatusExplanation explanation = new StatusExplanation(summary);
+
+    return taskResultBuilder.context("deploymentMonitorReasons", explanation).build();
+  }
+
+  String getRetrofitLogMessage(Response response) {
+    if (response == null) {
+      return "<NO RESPONSE>";
+    }
+
+    String body = "";
+    String status = "";
+    String headers = "";
+
+    try {
+      status = String.format("%d (%s)", response.getStatus(), response.getReason());
+      body = CharStreams.toString(new InputStreamReader(response.getBody().in()));
+      headers =
+          response.getHeaders().stream().map(Header::toString).collect(Collectors.joining("\n"));
+    } catch (Exception unused) {
+      // Nothing to do
```

Comment:
why nothing to do? What kind of exception do you expect in here, IO stuff? What if there's a sneaky NPE too?

Reply:
I don't really expect an exception (since I am not sure what retrofit may fail to include). but you are right I should at least log it - what i am doing is not acceptable.
and if there is an exception, we log best effort